tools/taskfile-swc.js

```diff
@@ -0,0 +1,69 @@
+// Based on Next.js swc taskr file.
+// https://github.com/vercel/next.js/blob/5378db8f807dbb9ff0993662f0a39d0f6cba2452/packages/next/taskfile-swc.js
+
+const { transform } = require('@swc/core');
+const path = require('path');
+
+module.exports = function (task) {
+  task.plugin('swc', {}, function* (file, environment, { stripExtension } = {}) {
+    // Don't compile .d.ts
+    if (file.base.endsWith('.d.ts')) return;
+
+    const setting = {
+      output: 'build',
+      options: {
+        module: {
+          type: 'commonjs',
+        },
+        env: {
+          targets: {
+            node: '14.17.6',
+          },
+        },
+        jsc: {
+          loose: true,
+          parser: {
+            syntax: 'typescript',
+            dynamicImport: true,
+          },
+        },
+      },
+    };
+
+    const filePath = path.join(file.dir, file.base);
+    const inputFilePath = path.join(__dirname, filePath);
+    const outputFilePath = path.dirname(path.join(__dirname, setting.output, filePath));
+
+    const options = {
+      filename: path.join(file.dir, file.base),
+      sourceMaps: true,
+      sourceFileName: path.relative(outputFilePath, inputFilePath),
```

Comment:
not sure whether we need source maps for the tools. if not, maybe the build speed would be even faster?

Reply:
Speeds up by 4 milliseconds, so probably worth keeping source maps for full debugging support.